File: code_samples/lua/testing_unit_tests.lua
-- Testing: Unit Tests
-- AI/ML Training Sample

UnitTests = {}
UnitTests.__ index = UnitTests

function UnitTests:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function UnitTests:process(input)
    self.data = input
end

function UnitTests:getData()
    return self.data
end

function UnitTests:validate()
    return #self.data > 0
end

-- Example usage
local instance = UnitTests:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/go/testing_unit_tests.go
package main

import (
    "fmt"
)

// Testing: Unit Tests
// AI/ML Training Sample

type UnitTests struct {
    Data string
}

func NewUnitTests() *UnitTests {
    return &UnitTests{
        Data: "",
    }
}

func (s *UnitTests) Process(input string) {
    s.Data = input
}

func (s *UnitTests) Validate() bool {
    return len(s.Data) > 0
}

func (s *UnitTests) GetData() string {
    return s.Data
}

func main() {
    instance := NewUnitTests()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
